tests/test_cli.py

```
@@ -2360,3 +2360,71 @@ def test_runtime_precondition_trap_non_json(
+        path = tmp_path / "input.vera"
+        path.write_text(self.SIMPLE_PROGRAM)
+        rc = cmd_run(str(path))
+        assert rc == 0
+        captured = capsys.readouterr()
+        assert "42" in captured.out
+
+    def test_check_reads_source_once(self) -> None:
+        """cmd_check parses from the already-read source, not by re-opening."""
+        # If check re-opened the file it would still work for a normal file,
+        # so we verify the behaviour via subprocess on /dev/stdin instead.
+        result = subprocess.run(
```

Ruff (0.15.7)

[error] `subprocess` call: check for execution of untrusted input

(S603)